### PR TITLE
feat: add getFeatureVariableJSON and getAllFeatureVariables apis

### PR DESCRIPTION
**Pull request by jaeopt**
## Summary
- add new APIs (getFeatureVariableJSON, getAllFeatureVariables)
- add new notification type (all-feature-variables)

### CODE REVIEW

File: core-api/src/main/java/com/optimizely/ab/Optimizely.java [hunk 1]
```diff
@@ -671,6 +711,10 @@ public String getFeatureVariableString(@Nonnull String featureKey,
         }
 
         Object convertedValue = convertStringToType(variableValue, variableType);
+        Object notificationValue = convertedValue;
```
[the remark this answers is not in the document]
Comment (jaeopt) [reply core-api/src/main/java/com/optimizely/ab/Optimizely.java:714]: We need notification map type different from the return OptimizelyJSON type.

File: core-api/src/main/java/com/optimizely/ab/Optimizely.java [hunk 2]
```diff
@@ -722,6 +768,103 @@ Object convertStringToType(String variableValue, String type) {
...
+            featureEnabled = variation.getFeatureEnabled();
+        } else {
+            logger.info("User \"{}\" was not bucketed into any variation for feature flag \"{}\". " +
+                    "The default values are being returned.", userId, featureKey);
+        }
+
+        Map<String, Object> valuesMap = new HashMap<String, Object>();
+        for (FeatureVariable variable : featureFlag.getVariables()) {
+            String value = variable.getDefaultValue();
+            if (featureEnabled && variation != null) {
```
[the remark this answers is not in the document]
Comment (jaeopt) [reply core-api/src/main/java/com/optimizely/ab/Optimizely.java:839]: Removed. A good catch.

File: core-api/src/main/java/com/optimizely/ab/Optimizely.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (pawels-optimizely) [reply core-api/src/main/java/com/optimizely/ab/Optimizely.java:714]: according to the comment, convertStringToType() is used only for notification listener. You already have  a case there for json type, can you return a map there ?

File: core-api/src/main/java/com/optimizely/ab/config/parser/ConfigParser.java [hunk 3]
```diff
@@ -16,6 +16,10 @@
  */
 package com.optimizely.ab.config.parser;
 
+import com.google.gson.JsonElement;
+import com.google.gson.JsonIOException;
+import com.google.gson.JsonSyntaxException;
+import com.google.gson.stream.JsonWriter;
```
Comment (mikecdavis) [inline core-api/src/main/java/com/optimizely/ab/config/parser/ConfigParser.java:22]: Why were these imports added?

File: core-api/src/main/java/com/optimizely/ab/config/parser/GsonConfigParser.java [hunk 4]
```diff
@@ -18,6 +18,7 @@
 
 import com.google.gson.Gson;
 import com.google.gson.GsonBuilder;
+import com.google.gson.JsonSyntaxException;
```
Comment (mikecdavis) [inline core-api/src/main/java/com/optimizely/ab/config/parser/GsonConfigParser.java:21]: Unused import?

**Pull request merged**